src/extra/gpkg/wkb.jl

```diff
@@ -0,0 +1,163 @@
+
+## Reading WKB Geometry BLOB
+
+``` julia
+    io = IOBuffer(WKBGeometryBLOB)
+    # load byte order in order to transport geometry 
+    # easily between systems of different endianness
+    wkbByteOrder = read(io, UInt8)
+    # load simple feature wkb geometry types supported
+    wkbGeometryType = read(io, UInt32)
+    # wkb points do not have a `numPoints` field 
+    if wkbGeometryType != 1
+      # load number of geometries in geometry set
+      numWkbGeometries = read(io, UInt32)
+    end
+    # load in WKBGeometry that contain geometry values
+    # w/ double precision numbers in the coordinates
+    # that are also subject to byte order rules.
+    wkbEndianness = isone(wkbByteOrder) ? ltoh : ntoh
+    # Note that Julia does not convert the endianness for you.
+    wkbGeometryBlob = wkbEndianness(read(io, Vector{UInt8}))
+```
+
+- Extended WKB allows applications to optionally add extra dimensions, and optionally embed an SRID
+ 99-402 was a short-lived extension to SFSQL 1.1 that used a high-bit flag
+to indicate the presence of Z coordinates in a WKB geometry.
+ When the optional wkbSRID is added to the wkbType, an SRID number is inserted after the wkbType number.
+⚠ This optional behaviour is not supported and will likely fail loading this variant
+
+- ISO WKB allows for higher dimensional geometries.
+SQL/MM Part 3 and SFSQL 1.2 use offsets of 1000 (Z), 2000 (M) and 3000 (ZM)
+⚠ only offsets of 1000 are recognized and supported and will likely fali loading this variant
+
+Other systems like GDAL supports three wkbVariants deviated from the wkbStandard
+- wkbVariantOldOgc:: Old-style 99-402
+- wkbVariantIso:: SFSQL 1.2 and ISO SQL/MM Part 3 
+- wkbVariantPostGIS1::PostGIS 1.X
+PostGIS supports a wider range of types (for example, CircularString, CurvePolygon)
+
+GeoIO GeoPackage reader supports wkbGeometryType using the SFSQL 1.2 use offset of 1000 (Z) and SFSQL 1.1 that used a high-bit flag, restricting some optional features
+
+GeoIO GeoPackage writer supports X,Y,Z coordinate offset of 1000 (Z) for wkbGeometryType
+
+## Example
+
+``` julia
+    geoms = []
+    io = IOBuffer
+    for row in wkbGeometryColumn
+      wkbbyteswap = isone(read(io, UInt8)) ? ltoh : ntoh
+      wkbtype = read(io, UInt32)
+      crs = LatLon{WGS84Latest}
+      haszextent = false
+      push!(geoms, gpkgwkbgeom(io, crs, wkbtype, haszextent, wkbbyteswap))
+    end
+``` 
+
+"""
+
+# According to https://www.geopackage.org/spec/#r20
+# GeoPackage SHALL store feature table geometries with the basic simple feature geometry types.
+# Geometry Types (Normative): https://www.geopackage.org/spec140/index.html#geometry_types
+# Note: this implementation supports (Core) Geometry Type Codes
+function gpkgwkbgeom(io, crs, wkbtype, zextent, bswap)
+  if wkbtype > 3
+    # 4 - 7 [MultiPoint, MultiLinestring, MultiPolygon, GeometryCollection]
+    elems = wkbmulti(io, crs, zextent, bswap)
+    Multi(elems)
```

Comment:
If a function is named `wkbmulti` it should return a `Multi` object. Please refactor to avoid this extra line where Multi is created manually.